packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostView.java

```diff
@@ -331,9 +334,34 @@ public boolean onKey(DialogInterface dialog, int keyCode, KeyEvent event) {
       mDialog.show();
       if (context instanceof Activity) {
         if (Build.VERSION.SDK_INT > Build.VERSION_CODES.R) {
-          int appearance =
-              ((Activity) context).getWindow().getInsetsController().getSystemBarsAppearance();
-          mDialog.getWindow().getInsetsController().setSystemBarsAppearance(appearance, appearance);
+          Window window = ((Activity) context).getWindow();
+          WindowInsetsController windowInsetsController = window.getInsetsController();
+          WindowInsetsController dialogWindowInsetsController = mDialog.getWindow().getInsetsController();
+
+          int appearance = windowInsetsController.getSystemBarsAppearance();
+          dialogWindowInsetsController.setSystemBarsAppearance(appearance, appearance);
+          dialogWindowInsetsController.setSystemBarsBehavior(windowInsetsController.getSystemBarsBehavior());
+
+          boolean statusBarsIsVisible = window.getDecorView().getRootWindowInsets().isVisible(WindowInsetsCompat.Type.statusBars());
+          if (statusBarsIsVisible) {
+            dialogWindowInsetsController.show(WindowInsetsCompat.Type.statusBars());
```

Comment:
Could we avoid these `if`s and just set the dialog's properties the same as the window?

Reply:
@lunaleaps, Sorry I'm a JS developer, I'm not an expert in JAVA and Android development, the fix I proposed is all I could do, I'll be thankful if you could mention someone with more expertise to fix the issue with best practice.